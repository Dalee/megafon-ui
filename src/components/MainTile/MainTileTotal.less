@import '~styles/base.less';

@b: e('.main-tile-total');

@{b} {
    flex-shrink: 0;
    width: 200px;
    @media @mobileS {
        margin: 0 auto;
    }
    @media @mobileB {
        width: 165px;
    }
    @media @desktop {
        width: 196px;
    }
    @media @desktopMU {
        width: 268px;
    }
    &__wrap {
        text-align: center;
        @media @desktopMU {
            width: 200px;
        }
    }
    &__cost {
        font-size: 18px;
        line-height: 24px;
        font-weight: 500;
        letter-spacing: 0.5px;
        margin-bottom: 8px;

        @media @desktopMU {
            font-size: 20px;
            line-height: 28px;
        }

        &_old-value {
            @{b}__price {
                font-size: 28px;
                font-weight: bold;
            }
        }
    }

    &__old-price {
        position: relative;
        &:before {
            content: '';
            position: absolute;
            left: 0;
            right: 0;
            top: 10px;
            border-bottom: 2px solid @fury;
        }
    }

    &__buttons {
        .b-link {
            width: 100% !important;
            margin-bottom: 0 !important;
            margin-right: 0 !important;
            padding-left: 0 !important;
            padding-right: 0 !important;
        }
        .b-link + .b-link {
            margin-top: 12px;
        }
        .b-link_view_with-subtitle {
            padding-top: 0;
            .b-link__title {
                margin-bottom: 0;
            }
        }
    }
}

