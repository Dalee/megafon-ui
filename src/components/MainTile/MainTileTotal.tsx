import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate } from '../../utils/cn';
import './MainTileTotal.less';
import Button from '../Button/Button';

interface IMainTileTotalProps {
    /** Payment
     * Object with args: value: string(required), unit: string(required), oldValue: string
     */
    payment: {
        value: string;
        unit: string;
        oldValue?: string;
    };
    /** Info - any object  */
    info?: {};
    /** Submit text */
    submitText?: string;
    /** Submit hander
     * Show button if included
     */
    handleSubmit?(e: React.SyntheticEvent<EventTarget>, info: {}): void;
}

const cn = cnCreate('main-tile-total');
class MainTileTotal extends React.Component<IMainTileTotalProps, {}> {
    static propTypes = {
        payment: PropTypes.shape({
            value: PropTypes.string.isRequired,
            unit: PropTypes.string.isRequired,
            oldValue: PropTypes.string,
        }).isRequired,
        info: PropTypes.object,
        handleSubmit: PropTypes.func,
    };

    static defaultProps: Partial<IMainTileTotalProps> = {
        submitText: 'Выбрать',
    };

    handleSubmit = (e: React.SyntheticEvent<EventTarget>): void => {
        const { handleSubmit, info } = this.props;

        handleSubmit && handleSubmit(e, info!);
    }

    render() {
        const { payment, handleSubmit, submitText } = this.props;

        return (
            <div className={cn('')}>
                <div className={cn('wrap')}>
                    <div className={cn('cost', { 'old-value': !!payment.oldValue })}>
                        <span className={cn('price')}>{payment.value}</span>&nbsp;
                        {payment.oldValue &&
                            <React.Fragment>
                                <span className={cn('old-price')}>{payment.oldValue}</span>
                                <br />
                            </React.Fragment>
                        }
                        <span className={cn('unit')}>{payment.unit}</span>
                    </div>
                    {handleSubmit &&
                        <div className={cn('buttons')}>
                            <Button
                                className={cn('button')}
                                onClick={this.handleSubmit}
                                width="full"
                            >
                                {submitText}
                            </Button>
                        </div>
                    }
                </div>
            </div>
        );
    }
}

export default MainTileTotal;
