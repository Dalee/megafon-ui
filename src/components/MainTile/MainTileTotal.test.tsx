import * as React from 'react';
import { shallow } from 'enzyme';
import MainTileTotal from './MainTileTotal';

const props = {
    payment: {
        value: '600 ₽',
        oldValue: '750 ₽',
        unit: 'в месяц',
    },
};

describe('<MainTileTotal />', () => {
    it('it renders MainTileTotal', () => {
        const wrapper = shallow(<MainTileTotal {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<MainTileTotal />', () => {
    it('it handle handleSubmit prop', () => {
        const handleSubmit = jest.fn();
        const wrapper = shallow(<MainTileTotal {...props} handleSubmit={handleSubmit} />);

        wrapper.find('.main-tile-total__button').simulate('click', {} as React.SyntheticEvent);
        expect(handleSubmit).toHaveBeenCalledTimes(1);
    });
});
