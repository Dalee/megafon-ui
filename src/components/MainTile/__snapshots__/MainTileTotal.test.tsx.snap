// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainTileTotal /> it renders MainTileTotal 1`] = `
<div
  className="main-tile-total"
>
  <div
    className="main-tile-total__wrap"
  >
    <div
      className="main-tile-total__cost main-tile-total__cost_old-value"
    >
      <span
        className="main-tile-total__price"
      >
        600 ₽
      </span>
       
      <React.Fragment>
        <span
          className="main-tile-total__old-price"
        >
          750 ₽
        </span>
        <br />
      </React.Fragment>
      <span
        className="main-tile-total__unit"
      >
        в месяц
      </span>
    </div>
  </div>
</div>
`;
